platforms/core-runtime/launcher/src/integTest/groovy/org/gradle/launcher/BuildEnvironmentIntegrationTest.groovy

```diff
@@ -110,6 +110,31 @@ task check {
         succeeds 'check'
     }
 
+    @Requires(IntegTestPreconditions.NotEmbeddedExecutor)
+    def "default keystore type can be set via custom security properties to #keystoreType"() {
+        def customPropertiesFile = file("custom-security.properties") << """
+        keystore.type=${keystoreType}
+        """
+
+        buildFile """
+        assert System.getProperty("java.security.properties") == "${customPropertiesFile.absolutePath}"
+        println "keystore.type = \${java.security.Security.getProperty("keystore.type")}"
+        println "default keystore type = \${java.security.KeyStore.getDefaultType()}"
```

Comment:
Is it necessary to check both of these? It looks like KeyStore.getDefaultType is calling Security.getProperty too.

Instead of checking the output, could we just check the value with an assert?

Lastly, you could drop the escaping of `$` by using +:
```suggestion
        println "keystore.type = " + java.security.Security.getProperty("keystore.type")
        println "default keystore type = " + java.security.KeyStore.getDefaultType()
```